storage/series/buffer.go

```diff
@@ -365,15 +367,28 @@ func (b *dbBuffer) Snapshot(ctx context.Context, blockStart time.Time) [][]xio.S
 			return
 		}
 
+		// We need to merge all the bootstrapped blocks / encoders into a single stream for the sake
+		// of being able to persist it to disk as a single encoded stream.
+		bucket.merge()
+
 		// This operation is safe because all of the underlying resources will respect the
 		// lifecycle of the context in one way or another. The "bootstrapped blocks" that
 		// we stream from will mark their internal context as dependent on that of the passed
 		// context, and the Encoder's that we stream from actually perform a data copy and
 		// don't share a reference.
-		res = append(res, bucket.streams(ctx))
+		streams := bucket.streams(ctx)
+		if len(streams) > 1 {
```

Comment:
Maybe for just a tiny bit more safety (in case the merge method does something really weird and leaves us with no data at all in the bucket due to an error) we can do `if len(streams) != 1 { return errNotOneStreamAfterMerge }`?

Reply:
sure